pkg/codegen/python/gen.go

```diff
@@ -1528,14 +1539,24 @@ func (mod *modContext) genMethods(w io.Writer, res *schema.Resource) {
 		}
 
 		// Now simply call the function with the arguments.
-		var typ, ret string
+		var typ string
 		if retTypeNameQualified != "" {
 			// Pass along the private output_type we generated, so any nested output classes are instantiated by
 			// the call.
 			typ = fmt.Sprintf(", typ=%s", retTypeNameQualified)
-			ret = "return "
 		}
-		fmt.Fprintf(w, "        %spulumi.runtime.call('%s', __args__, res=__self__%s)\n", ret, fun.Token, typ)
+
+		if method.Function.Outputs == nil {
+			fmt.Fprintf(w, "        pulumi.runtime.call('%s', __args__, res=__self__%s)\n", fun.Token, typ)
+		} else if shouldLiftReturn {
+			// Store the return in a variable and return the property output
+			fmt.Fprintf(w, "        res = pulumi.runtime.call('%s', __args__, res=__self__%s)\n", fun.Token, typ)
+			fmt.Fprintf(w, "        return res.%s\n", PyName(fun.Outputs.Properties[0].Name))
```

Comment:
Nit: It's possible (however unlikely) that someone could have `res` as a parameter name. Using an even more unlikely name like `__result__` would help mitigate.

No need to change it in your PR. I'm going to follow-up soon with a change that adds some tests and adds .NET, and while doing that, I'll consider tweaking this.

Reply:
Yes, that's a good point and not one I'd considered. A prefix is definitely a better idea here.